### PR TITLE
feat: Complete the node initialization logic

### PR DESCRIPTION
**Pull request by ssongliu**

### CODE REVIEW

File: agent/app/service/backup.go [hunk 1]
```diff
@@ -351,24 +376,7 @@ func NewBackupClientMap(ids []string) (map[string]backupClientHelper, error) {
...
-		if err != nil {
-			return nil, err
-		}
-		if err := json.Unmarshal(item, &accounts); err != nil {
-			return nil, fmt.Errorf("err response from master: %v", data)
-		}
+		accounts, _ = backupRepo.List(commonRepo.WithByIDs(idItems))
 	}
 	clientMap := make(map[string]backupClientHelper)
 	for _, item := range accounts {
```
Comment (wanghe-fit2cloud) [inline agent/app/service/backup.go:382]: The provided code is missing the implementation functions for 'Sync' (function takes an operation, data as parameters), 'SearchRecordsWithPage', 'LoadBackupOptions()', and 'searchRecordsByCronjobWithPage'. These methods could contain optimizations like using caching to avoid unnecessary database queries, use a different structure if operations may have variable number of arguments or results.

Additionally, some fields like:

* `accessKey` and `credential` should be strings instead of byte arrays
* Account's ID value needs more specific handling based on its purpose

However, without seeing actual content of these methods there aren't much changes needed except those mentioned.

It might also be useful to implement error handling appropriately where it occurs in order to handle all possible errors correctly.

For the given information, here is the most relevant advice and adjustments that can be made.

* Implement the Sync function with proper validation for operation parameter types.
* Use better naming conventions for methods names and variables.
* Include appropriate error codes and messages in method failure scenarios.
* Make sure input checking properly handles edge cases and unexpected inputs.

This approach allows improving maintainability and readability while keeping the core functionality intact.

File: core/app/service/setting.go [hunk 2]
```diff
@@ -517,17 +466,3 @@ func checkCertValid() error {
...
-	if len(matches) <= 0 {
-		return ""
-	}
-	var port string
-	port = matches[3]
-	if len(port) != 0 {
-		return port
-	}
-	return ""
-}
```
Comment (wanghe-fit2cloud) [inline core/app/service/setting.go:533]: There do not seem to be any specific changes or discrepancies mentioned in terms of code quality, complexity, performance, security vulnerabilities, etc. However, there are some common conventions such as removing unused imports at the top level, organizing imports appropriately according to Go's naming convention, and ensuring consistent use of functions within their own class or module rather than outside it.

For instance:

```
func getSomeValue() int { ... } // put inside this function instead of being out side
``` 

The code is generally well-structured but might benefit slightly from more explicit checks like using defer statements when you need to perform tasks that will continue after the block ends. This helps ensure those cleanup operations happen before the code executes. Other small refinements include adding spaces around operators and comments where appropriate, capitalizing variable names consistently, ensuring meaningful parameter lists for loops and other sections involving iteration. In the context of Golang standard library modules, `import` directives should ideally follow conventional import style.

In summary, there doesn't appear to be major defects or areas requiring modifications except possibly improvements in documentation about usage, clarity, readability, and consistency which could greatly enhance future maintenance efforts.

File: agent/app/repo/app_launcher.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	return global.DB.Save(launcher).Error
+}
+
+func (u *LauncherRepo) Delete(opts ...DBOption) error {
+	db := global.DB
+	for _, opt := range opts {
+		db = opt(db)
+	}
+	return db.Delete(&model.AppLauncher{}).Error
+}
```
Comment (wanghe-fit2cloud) [inline agent/app/repo/app_launcher.go:55]: I'm sorry, but you didn't provide any code to review. Please send the updated version of the code so that I can analyze it.
